lib/Parse/ParseExpr.cpp

```diff
@@ -1697,6 +1697,10 @@ ParserResult<Expr> Parser::parseExprPrimary(Diag<> ID, bool isExprBasic) {
   case tok::l_square:
     return parseExprCollection();
 
+  case tok::pound_assert:
+    return parseExprPoundAssert();
+    break;
```

Comment:
`#assert()` should really be modeled as a expression? From Sema code, I understand the type of `#assert(...)` expression is always `Void`. But I don't think `let foo: Void = #assert(true)` should even be accepted.
Why don't you model it as a statement?